### PR TITLE
Implement some LGPE exclusive moves and other moves and modernise some weathers

### PR DESCRIPTION
**Pull request by Aeliko**
Moves:
- [x] Zippy Zap
- [x] Bouncy Bubble
- [x] Sizzly Slide
- [x] Glitzy Glow
- [x] Baddy Bad
- [x] Sappy Seed
- [x] Freezy Frost
- [x] Sparkly Swirl
- [ ] Pika Papow (current effect is correct, but missing message saying "\<name\>'s sure-hit move!" and Joycon motion activation method, not doing it for now)
- [ ] Veevee Volley (current effect is correct, but missing message saying "\<name\>'s sure-hit move!" and Joycon motion activation method, not doing it for now)
- [x] Diamond Storm
- [x] Geomancy (after the absorbing power message, the previous message will be displayed again, no idea how to fix)
- [x] Snowscape
- [x] Psychic Noise

Text:
- [x] Update Light Screen text, seems it is hardcoded to say Light Screen in modern games (see Glitzy Glow)
- [x] Update Reflect text, seems it is hardcoded to say Reflect in modern games (see Baddy Bad)
- [x] Update Heal Block text, ~~seems it is hardcoded to say Heal Block in modern games~~ (see Psychic Noise, already handled in vanilla)

Weather:
- [x] Snow (missing animation)

Abilities
(There are some issues with printing messages I have no idea how to solve)
- [x] Snow Warning #151 (Supports both Hail and Snow via `config.s`)
- [x] Drought #151
- [x] Drizzle #151
- [x] Sandstorm #151

Stuff to do in the future
- [x] Moonlight, Morning Sun, Synthesis ([`BtlCmd_WeatherHPRecovery`](https://github.com/pret/pokeheartgold/blob/2b669d0919dca73ed77192144c7bcfcc0ff85a12/src/battle/battle_command.c#L3647) Snow causes Moonlight, Synthesis, and Morning Sun to recover ¼ of the user's maximum HP.)
- [x] Weather Ball (`weatherballdamagecalc`)
- [x] Replace every instance of `sp->moveTbl[moveno].split` with `GetMoveSplit(sp, moveno)` (Not doing it for now because not sure if the `LONG_CALL` stuff has negative effects, and this only solves 1 edge case)
- [x] Blizzard accuracy (`BattleSystem_CheckMoveEffect`) (Code is done, missing hook)

### CODE REVIEW

**Comment by BluRosie** [comment]
do you intend to complete these tasks or do you want me to take a look at some point?

**Comment by BluRosie** [comment]
the `LONG_CALL` stuff for battle code is only necessary if it will be used in the individual code overlays (damage calc and such).  it would have no negative effect--it just changes the way that the codegen links it to everything else

**Comment by BluRosie** [comment]
should be fine to hook `BattleSystem_CheckMoveEffect` as well, just need to do the whole r5 hook with a `bytereplacement`--see [`CantEscape`](https://github.com/search?q=repo%3ABluRosie%2Fhg-engine%20CantEscape&type=code)
image

File: src/individual/CalcBaseDamage.c [hunk 1]
```diff
@@ -755,7 +755,7 @@ int CalcBaseDamage(void *bw, struct BattleStruct *sp, int moveno, u32 side_cond,
         {
             sp_defense = sp_defense * 15 / 10;
         }
-        if ((field_cond & WEATHER_HAIL_ANY) &&
+        if ((field_cond & WEATHER_SNOW_ANY) &&
```
[the remark this answers is not in the document]
Comment (Aeliko) [reply src/individual/CalcBaseDamage.c:758]: In [Generation IX](https://bulbapedia.bulbagarden.net/wiki/Generation_IX), snow replaces hail. It lasts for 5 turns if created using [Snow Warning](https://bulbapedia.bulbagarden.net/wiki/Snow_Warning_(Ability)) or [Snowscape](https://bulbapedia.bulbagarden.net/wiki/Snowscape_(move)), or 8 turns if created by a Pokémon holding an [Icy Rock](https://bulbapedia.bulbagarden.net/wiki/Icy_Rock). The move [Chilly Reception](https://bulbapedia.bulbagarden.net/wiki/Chilly_Reception_(move)) also creates snow and switches the user out. It raises the [Defense](https://bulbapedia.bulbagarden.net/wiki/Stat#Defense) stat of all [Ice-type](https://bulbapedia.bulbagarden.net/wiki/Ice_(type)) Pokémon by 50%. Unlike hail, snow does not damage non-Ice types.

File: armips/move/battle_sub_seq/358.s [hunk 2]
```diff
@@ -14,7 +14,6 @@
 a001_358:
     changevar2 VAR_OP_SET, VAR_HP_TEMP, VAR_HIT_DAMAGE
     if IF_EQUAL, VAR_HP_TEMP, 0x0, _0030
-    // removed the command to half the value
 _0030:
```
Comment (BluRosie) [inline armips/move/battle_sub_seq/358.s:17]: to be clear, when this if passes it still just jumps direct to `_0030`--no need to have the if

**Review by BluRosie — COMMENTED** [review]
looks pretty good to me

will have to go through and add all the other tasks soon

**Comment by BluRosie** [comment]
will have to fix geomancy sometime this weekend and then merge this as good

**Review by BluRosie — APPROVED** [review]

**Comment by BluRosie** [comment]
i will update the project when i get home

**Pull request merged**